pandas/tests/dtypes/test_inference.py

```diff
@@ -288,7 +288,10 @@ class MockFile:
     assert not is_file(data)
 
 
-@pytest.mark.parametrize("ll", [collections.namedtuple("Test", list("abc"))(1, 2, 3)])
+test_tuple = collections.namedtuple("Test", ["a", "b", "c"])
```

Comment:
Out of curiosity is there any issue you saw that suggested doing this? I don't think its evident why this would need to exist without knowing the context of this PR.

If it is an issue with mypy or typeshed would sometimes prefer to wait for a fix upstream; generally not in a rush with these

Reply:
mypy does not infer that the `collections.namedtuple()` call returns a class of user-defined type. It, for some reason, assumes that it returns a tuple. `tuple()` expects an argument of Iterable[Any], which is why we see the error `pandas\tests\dtypes\test_inference.py:291: error: Argument 1 to "tuple" has incompatible type "int"; expected "Iterable[Any]"`.

mypy also does not seem to infer that `list('abc')` is a literal of type List[str]. The error doesn't show up if the above one isn't corrected, but if you change `['a', 'b', 'c']` to `list('abc'), then you get the error `pandas/tests/dtypes/test_inference.py:291: error: List or tuple literal expected as the second argument to namedtuple()`. mypy does infer that `['a', 'b', 'c']` is of type List[str].

IMO these do seem to be limitations in mypy.

Reply:
OK thanks for confirming. It would be good for future edits in this project and the entire python eco-system as a whole if those were raised as issues with MyPy, if not already

Not saying this was a problem merging, but especially for internal-only annotations like this there is no rush. If something like that is raised with MyPy and looks like it would be available in a few releases would rather wait than making code edits here

Reply:
Opened two issues with typeshed for
1. incorrect type inference on `collections.namedtuple()`: https://github.com/python/typeshed/issues/3394#issue-510165256
2. incorrect type inference for literal `list('abc')`: https://github.com/python/typeshed/issues/3395#issue-510169457

Reply:
Opened two issues with [typeshed](https://github.com/python/typeshed) -
1. `namedtuple()` inferred as type tuple: https://github.com/python/typeshed/issues/3394
2. `list('abc')` not inferred as type List[str]: https://github.com/python/typeshed/issues/3395